File: src/main/scala/com/lightbend/akka/johan/stream/samples/scala/Sample1.scala
/*
 * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
 */
package com.lightbend.akka.johan.stream.samples.scala

import akka.NotUsed
import akka.actor.ActorSystem
import akka.stream.ActorMaterializer
import akka.stream.scaladsl.{Flow, Sink, Source}

/**
  * Minimal sample of Akka Streams
  */
object Sample1 extends App {
  implicit val system = ActorSystem()
  implicit val mat = ActorMaterializer()

  val source = Source(0 to 20000000)

  val flow = Flow[Int].map(_.toString())

  val sink = Sink.foreach[String](println(_))

  val runnable = source.via(flow).to(sink)

  runnable.run()

}

File: src/main/java/com/lightbend/akka/johan/stream/samples/java/Sample1.java
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
 */
package com.lightbend.akka.johan.stream.samples.java;

import akka.Done;
import akka.NotUsed;
import akka.actor.ActorSystem;
import akka.stream.*;
import akka.stream.javadsl.*;

import java.util.concurrent.CompletionStage;

/**
 * Minimal sample of Akka Streams
 */
public class Sample1 {
    public static void main(String[] args) {
        final ActorSystem system = ActorSystem.create();
        final Materializer materializer = ActorMaterializer.create(system);

        final Source<Integer, NotUsed> source =
                Source.range(0, 20_000_000);

        final Flow<Integer, String, NotUsed> flow =
                Flow.fromFunction((Integer n) -> n.toString());

        final Sink<String, CompletionStage<Done>> sink =
                Sink.foreach(str -> System.out.println(str));

        final RunnableGraph<NotUsed> runnable = source.via(flow).to(sink);

        runnable.run(materializer);
    }
}
